frontend/components/PhoneinputTest.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { Component, useRef } from "react";
+import {
+  SafeAreaView,
+  StyleSheet,
+  View,
+  StatusBar,
+  TouchableOpacity,
+  Text,
+  Pressable,
+} from 'react-native';
+import PhoneInput from 'react-native-phone-number-input';
+// import {Colors} from 'react-native/Libraries/NewAppScreen';
+// const phoneInput = useRef<PhoneInput>(null);
+type State = {
+    value: string,
+    countryCode: string,
+    formattedValue: string,
+    valid:boolean,
+    disabled: boolean,
+    showMessage: boolean,
+
+}
+class PhoneinputTest extends Component<{}, State> {
+    phoneInput=React.createRef<PhoneInput>();
+    constructor(props: {}) {
+      super(props);
+      this.state = {
+        value: '',
+        countryCode: '',
+        formattedValue: '',
+        valid: false,
+        disabled: false,
+        showMessage: false,
+      }
+    }
+  demo = () =>{
+      console.log("demo triggered")
+  }
+  
+  render(){
+    return (
+        <>
+          <StatusBar barStyle="dark-content" />
+          <View style={styles.container}>
+            <SafeAreaView style={styles.wrapper}>
+              {this.state.showMessage && (
+                <View style={styles.message}>
+                  <Text>Country Code : {this.state.countryCode}</Text>
+                  <Text>Value : {this.state.value}</Text>
+                  <Text>Formatted Value : {this.state.formattedValue}</Text>
+                  <Text>Valid : {this.state.valid ? 'true' : 'false'}</Text>
+                </View>
+              )}
+              <PhoneInput
+                ref={this.phoneInput}
+                defaultValue={this.state.value}
+                defaultCode="IN"
+                layout="first"
+                onChangeText={(text) => { this.setState({value: text}) }}
+                onChangeFormattedText={(text) => {
+                this.setState({formattedValue: text})
+                this.setState({countryCode: this.phoneInput.current?.getCountryCode() || ''})
+                }}
+                countryPickerProps={{withAlphaFilter:true}}
+                disabled={this.state.disabled}
+                withDarkTheme
+                withShadow
+                autoFocus
+              />
+              <TouchableOpacity
+                style={styles.button}
+                onPress={() => {
+                  const checkValid = this.phoneInput.current?.isValidNumber(this.state.value);
+                  this.setState({showMessage: true})
+                  this.setState({valid: checkValid ? checkValid : false})
```

Comment:
think we can simply pass `checkValid`?

Reply:
yes. The entire PhoneinputTest.tsx file is an example file of react-native-phone-input. I just used it to test how the code works. But it is not properly integrated to support web.